src/main/java/com/ibm/devops/connect/CloudSocketComponent.java

```diff
@@ -161,11 +163,26 @@ public void handleDelivery(String consumerTag, Envelope envelope,
                 }
             };
 
-            channel.basicConsume(queueName, true, consumer);
+            if (queueIsAvailable(channel, queueName)) {
+                log.info("Queue is available, consuming...");
+                channel.basicConsume(queueName, true, consumer);
+            } else {
+                log.info("Queue not available, waiting to attempt reconnect");
+                connectToAMQP();
```

Comment:
I thought this was getting removed to so it just relies on the reconnectExecutor